src/User/UserContainer.jsx

```diff
@@ -1,53 +1,99 @@
+import { noManager } from "../Constants/Constants";
 import ProfilePage from "./ProfilePage";
 import UpdateProfile from "./UpdateProfile";
 import { getUserById } from "../api.js";
 
-const UserContainer = ({ sessionToken }) => {
-	const [user, setUser] = useState(null);
-	const [isLoading, setIsLoading] = useState(true);
-	const [error, setError] = useState(false);
-	const { id } = useParams();
+const UserContainer = () => {
+  const [user, setUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [managerName, setManagerName] = useState("");
+  const { id } = useParams();
   const [cookies] = useCookies([cookieName]);
-    
-	const fetchData = async () => {
-		setIsLoading(true);
-
-		try {
-			const user = await getUserById(id, cookies.sessionToken.token);
-			setUser(user);
-		} catch (error) {
-			setError(true);
-		}
-		setIsLoading(false);
-	};
-
-	useEffect(() => {
-		if (id) {
-			fetchData(id);
-		}
-	}, [id]);
-
-	return (
-		<Switch>
-			<Route path="/users/:id/edit">
-				<UpdateProfile
-					user={user}
-					id={id}
-					setUser={setUser}
-					isLoading={isLoading}
-					error={error}
-					sessionToken={sessionToken}
-				/>
-			</Route>
-			<Route path="/users/:id">
-				<ProfilePage user={user} isLoading={isLoading} error={error} />
-			</Route>
-		</Switch>
-	);
+
+  const isCurrentUser = () => {
+    return Number(id) === cookies.sessionToken.userId;
+  };
+
+  const getManagerName = async user => {
+    if (user) {
+      if (user.managerId) {
+        try {
+          const manager = await getUserById(
+            user.managerId,
+            cookies.sessionToken.token
+          );
+          setManagerName(`${manager.firstName} ${manager.lastName}`);
+        } catch (error) {
+          setError(true);
+        }
+      } else {
+        setManagerName(noManager);
+      }
+    }
+  };
+
+  const fetchData = async userId => {
+    setIsLoading(true);
+
+    try {
+      const user = await getUserById(userId, cookies.sessionToken.token);
+      setUser(user);
+      setError(false);
+    } catch (error) {
+      setError(true);
+    }
+
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    if (id) {
+      fetchData(id);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id]);
+
+  useEffect(() => {
+    if (isCurrentUser()) {
+      getManagerName(user);
+    } else {
+      setManagerName("");
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isCurrentUser(), user ? user.id : null]);
+
+  return (
+    <Switch>
+      <Route path="/users/:id/edit">
+        {isCurrentUser() ? (
+          <UpdateProfile
+            user={user}
+            id={id}
+            setUser={setUser}
+            isLoading={isLoading}
+            error={error}
+            sessionToken={cookies.sessionToken}
+          />
+        ) : (
+          <Redirect to={`/users/${cookies.sessionToken.userId}`} />
```

Comment:
what happens if a user isn't logged in? probably best to just redirect to `"/users/:id` right

Reply:
You aren't allowed to see `/users/:id` if you aren't logged in though so possibly that needs to be redirected in app not user container

Reply:
I've added an extra layer of redirects which trigger when you're not logged in taking you away from the prohibited route to `/users/login` so then the other redirect can only ever be triggered when you are logged in and so `/users/${cookies.sessionToken.userId}` is a valid route.

Reply:
My impression was when you are viewing a user who is your employee you would still use this route of `user\:id`? It doesn't matter for now was just bringing to your attention

Reply:
That is the case, the redirect in `App` stops you viewing the `/user/:id` route if you aren't logged in, the redirect in `UserContainer` stops you viewing the `/user/:id/edit` unless you are the user being edited. So if you're a manager who is logged in you can still see your employees, its only if you aren't logged in that you can't see the `/users/:id` route.